packages/tooling/src/lib/parameters-to-json-schema.js

```diff
@@ -15,42 +17,110 @@ const types = {
 
 function getBodyParam(pathOperation, oas) {
   const schema = getSchema(pathOperation, oas);
-
   if (!schema) return null;
 
+  const cleanupSchemaDefaults = obj => {
+    Object.keys(obj).forEach(prop => {
+      if (typeof obj[prop] === 'object') {
+        cleanupSchemaDefaults(obj[prop]);
+      } else if (prop === 'default') {
+        if ('allowEmptyValue' in obj && obj.allowEmptyValue && obj[prop] === '') {
+          // If we have `allowEmptyValue` present, and the default is actually an empty string, let it through as it's
+          // allowed.
+        } else if (obj[prop] === '') {
+          delete obj[prop];
+        }
+      }
+    });
+
+    return obj;
+  };
+
   const type = schema.type === 'application/x-www-form-urlencoded' ? 'formData' : 'body';
 
   return {
     type,
     label: types[type],
-    schema: oas.components ? { definitions: { components: oas.components }, ...schema.schema } : schema.schema,
+    schema: oas.components
+      ? { definitions: { components: cleanupSchemaDefaults(oas.components) }, ...cleanupSchemaDefaults(schema.schema) }
+      : cleanupSchemaDefaults(schema.schema),
   };
 }
 
-function hasCommonParameters(pathOperation) {
+function getCommonParams(pathOperation) {
   const { path } = pathOperation || {};
-  const commonParams = ((((pathOperation || {}).oas || {}).paths || {})[path] || {}).parameters;
-  return !!(commonParams && commonParams.length !== 0);
+  if (pathOperation && 'oas' in pathOperation && 'paths' in pathOperation.oas && path in pathOperation.oas.paths) {
+    if ('parameters' in pathOperation.oas.paths[path]) {
+      return pathOperation.oas.paths[path].parameters;
+    }
+  }
+
+  return [];
 }
 
 function getOtherParams(pathOperation, oas) {
-  let pathParameters = pathOperation.parameters || [];
```

Comment:
Renamed this from `pathParameters` to `operationParameters` because in the context of what this is doing, it's building up an array of every parameter on the option; not an array of every **path** parameter on the option.

<sub>i spent a good number of minutes trying to hunt down some weird edge cases when writing tests before realizing this is what was going on 🙃 </sub>